lib/JMAP/Tester.pm

```diff
@@ -839,26 +875,99 @@ sub logout {
   Carp::confess("can't logout: no authentication_uri configured")
     unless $self->has_authentication_uri;
 
-  my $logout_res = $self->ua->delete(
-    $self->authentication_uri,
+  my $req = HTTP::Request->new(
+    DELETE => $self->authentication_uri,
     [
       'Content-Type' => 'application/json; charset=utf-8',
       'Accept'       => 'application/json',
     ],
   );
 
-  if ($logout_res->code == 204) {
-    $self->_access_token(undef);
+  my $future = $self->ua->request($self, $req, 'auth')->then(sub {
+    my ($res) = @_;
 
-    return JMAP::Tester::Result::Logout->new({
-      http_response => $logout_res,
-    });
-  }
+    if ($res->code == 204) {
+      $self->_access_token(undef);
 
-  return JMAP::Tester::Result::Failure->new({
-    ident => "failed to log out",
-    http_response => $logout_res,
+      return Future->done(
+        JMAP::Tester::Result::Logout->new({
+          http_response => $res,
+        })
+      );
+    }
+
+    return Future->fail(
+      JMAP::Tester::Result::Failure->new({
+        ident => "failed to log out",
+        http_response => $res,
+      })
+    );
   });
+
+  return $self->should_return_futures ? $future : $future->$Failsafe->get;
+}
+
+=method http_request
+
+  my $response = $jtest->http_request($http_request);
+
+Sometimes, you may need to make an HTTP request with your existing web
+connection.  This might be to interact with a custom authentication mechanism,
+to access custom endpoints, or just to make very, very specifically crafted
+requests.  For this reasons, C<http_request> exists.
+
+This method respects the C<should_return_futures> attributes of the
+JMAP::Tester object, and in futures mode will return a future that will resolve
+to the HTTP::Response.
```

Comment:
We document HTTP::Response here which is nice but we don't spell out that $http_requeset is an HTTP::Request object, and I think we should.

It'd be nice if we also L<...> wrapped them

Reply:
Updated.